File: EPortal/IMS.EverestPortal.API/Models/MarketDefinitionBaseMap.cs
using System;
using System.Collections.Generic;
using System.Linq;
using System.Runtime.Serialization;
using System.Web;

namespace IMS.EverestPortal.API.Models
{
    [DataContract]
    public class MarketDefinitionBaseMap
    {
        [DataMember]
        public int Id { get; set; }
        [DataMember]
        public string Name { get; set; }
        [DataMember]
        public int MarketBaseId { get; set; }
        [DataMember]
        public virtual MarketBase MarketBase { get; set; }
        [DataMember]
        public virtual List<AdditionalFilter> Filters { get; set; }
        [DataMember]
        public string DataRefreshType { get; set; }
        [DataMember]
        public int MarketDefinitionId { get; set; }
        internal virtual MarketDefinition MarketDefinition { get; set; }
    }
}

File: EPortal/IMS.EverestPortal.Web/app/shared/models/market-definition-base-map.ts
import { AdditionalFilter } from './additional-filter';
import { MarketBase } from './market-base';

export class MarketDefinitionBaseMap {
    Id: number;
    Name: string;
    MarketBaseId:number;
    MarketBase: MarketBase;
    BaseFilters?:any[];
    Filters: AdditionalFilter[];
    DataRefreshType: string;
    Selected?:boolean;
    CurrentStatus?:string;
}

